### ISSUE TITLE
Feature request: Arena gamemode

### ISSUE THREAD

**Issue by jFransham** [x6herbius/hl2dmx#2]
Choose weaponset (Including random? Only 3 or 4 weapons per game) as convar, start with full ammo. On kill, drop all remaining ammo as single item (I'm thinking have the model be a melon). Automatically set our weird convars to whatever we set them to when we start a server up anyway. Cannot pick up new weapons.

**Comment by noodlecollie** [comment]
This sounds cool but it might be complicated to integrate it nicely, being a whole new game mode.

**Comment by jFransham** [comment]
Yep! I think the current game is already implemented as an instance of a gamemode class + a whole load of code in the player/weapon etc that assumes that you're playing in that game mode, so we'd have to factor out the stuff that we want to change into method calls to the gamemode and then put our custom code in that. I quite like your way of making all the weird stuff optional and off by default, so if we can avoid replacing the current game logic that'd be awesome.

Also, if we factor more things into a gamemode class we can add scripting support like 200x easier, and that's something I'd really like to try and do.
